### ISSUE TITLE
Extremely slow startup times since latest 1.4.x updates

### ISSUE THREAD

**Issue by brittyazel** [EmmyLua/IntelliJ-EmmyLua#548]
Since the latest round of updates, launching IntelliJ takes a very long time, with the IDE sitting at  "Lua indexing: waiting to start ..." and then "Scanning files to index", both taking a couple of minutes to complete. Likewise, while this is happening, it's using >15% of my CPU.

This is with version 1.4.2, but it was also present in the prior version.

**Comment by Meivyn** [comment]
Can confirm, with a very large amount of files, the IDE is completely freezing, and the CPU usage is stuck at 100%.

**Comment by tangzx** [comment]
Can you please try this version? 
https://github.com/EmmyLua/IntelliJ-EmmyLua/actions/runs/5206248623

**Comment by brittyazel** [comment]
I cannot install that version. I get an error saying "failed to load plugin descriptor from file"

**Comment by tangzx** [comment]
> I cannot install that version. I get an error saying "failed to load plugin descriptor from file"

You need to unzip it and you will see another zip file, use that file to install

**Comment by Meivyn** [comment]
This still freezes the IDE as soon as "Indexing dependencies" is done. 

It however doesn't freeze indefinitely like it used to, but file browsing is extremely laggy, and it consumes a lot of RAM.
It also still freezes after a while.

**Comment by tangzx** [comment]
@Meivyn I can't reproduce the problem, So ... I wonder if your project is complicated? How many Lua files are there and how the size are they?

**Comment by Meivyn** [comment]
It's not really a project - I am simply loading the whole `Interface/AddOns` folder from World of Warcraft. Those add-ons are written in Lua. I always did that without a problem until that update. But that implies that there are *a lot* of files, so I can't tell the size or how many of them though.

**Comment by tangzx** [comment]
@Meivyn I don't have WOW installed, where I can get the whole addons folder? Can you pack and upload it for me to test ?

**Comment by Meivyn** [comment]
Here it is: https://send.vis.ee/download/5a3bb95c9f5110f6/#87S-grQ_dqoljGGr6-rbvQ

**Comment by tangzx** [comment]
> I always did that without a problem until that update
Yes, I rewrote the `index system`,  since the `index system` in previous version  cannot work in IDEA231. I had to rewrite it.

**Comment by tangzx** [comment]
> Here it is: https://send.vis.ee/download/5a3bb95c9f5110f6/#87S-grQ_dqoljGGr6-rbvQ

Thanks

**Comment by tangzx** [comment]
> Here it is: https://send.vis.ee/download/5a3bb95c9f5110f6/#87S-grQ_dqoljGGr6-rbvQ

Sorry,but I cannot download it. Got an error:
![image](https://github.com/EmmyLua/IntelliJ-EmmyLua/assets/3354360/6b21e8cb-bdd5-4855-9f19-b28d2db8c1fd)

**Comment by Meivyn** [comment]
Hmm, no clue what that error means, I am not understanding this language. What about this one?
https://wormhole.app/4zyjp#mt4Ty_GWKYvwko_UeOUd4A

**Comment by tangzx** [comment]
@Meivyn  It's my problem, finally I downloaded it, thanks , I'll try it

**Comment by brittyazel** [comment]
It's a bit better. It doesn't lock up the UI nearly as much, though it takes 2-3min to finish it's analysis every single time I open my projects. It appears like it's having to analyze over 100k items according to the new loading status. Almost all of the things it is "analyzing" are the API files added to the project as external libraries. It seems odd that it has to do it every single time. Prior to 1.4.x it would open in just a few seconds.

**Comment by brittyazel** [comment]
I have the exact same situation as @Meivyn. We are both loading the whole Interface codebase from World of Warcraft. I have it added as a project SDK. It's quite a bit of files:
https://github.com/Gethe/wow-ui-source

**Comment by brittyazel** [comment]
Would it be possible to implement some sort of caching system for between sessions? I.e. the 100,000+ objects needing scanned in the World of Warcraft API aren't changing that often, and therefore it seems wasteful to scan them over and over again on each launch. Likewise, even when they are changed, it's really just a subset of them. If the session somehow knew which files had or had not been changed from one session to the next, and could use that information to inform the indexer, that could considerably speed things up.

**Comment by Meivyn** [comment]
I am getting an infinite indexing loop with the latest version... It starts to index the module, and then the SDK, and then the module, and then the SDK, and then the module...

**Comment by brittyazel** [comment]
@Meivyn Just a thought, have you tried invalidating your caches and restarting the IDE? I'm not able to replicate your issue on my end.

**Comment by Meivyn** [comment]
Yeah, I did, and that didn't help. It was fixed when I reverted back to 1.3.20.

**Comment by Meivyn** [comment]
@tangzx Would appreciate it if you could look at it/send some news since 2023.2 is now officially out.
